Test/git-issues/git-issue-1604g.dfy

```diff
@@ -0,0 +1,18 @@
+// RUN: %dafny /compile:1 /errorLimit:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+predicate EvenNat(n: nat) { n % 2 == 0 }
+predicate TrueInt(x: int) { true }
+
+method NatTypeInferenceType() {
+  // These behave as they had before
+  assert forall n: nat :: EvenNat(n) ==> TrueInt(n); // correct, since n is nat
+  assert forall x: int :: EvenNat(x) ==> TrueInt(x); // precondition violation, since EvenNat expects a nat and x is int
+  assert forall x: int :: 0 <= x && EvenNat(x) ==> TrueInt(x); // good
+  assert forall x: int :: EvenNat(x) && 0 <= x ==> TrueInt(x); // precondition violation (good)
+  assert forall n :: EvenNat(n) ==> TrueInt(n); // since n is inferred to be an int, an precondition violation is reported
```

Comment:
The comments on L12, L16, and L17 say that errors are reported, but the `.expect` file doesn't show any errors. Please update the comments.

Reply:
I'm not use if you are commenting the right file, because the expect file does show three errors:
https://github.com/dafny-lang/dafny/pull/1702/files#diff-3adf349d1fbe3d24aaf31d072ed9c82757730d1ed46b657bcbf074483ac27d96